src/envoy/mixer/http_filter.cc

```diff
@@ -257,6 +258,37 @@ class CheckData : public HttpCheckData,
     }
     return false;
   }
+
+  bool GetJWTPayload(
+      std::map<std::string, std::string>* payload) const override {
+    if (payload == nullptr) {
+      return false;
+    }
+    const std::string kSecIstioAuthUserinfo("sec-istio-auth-userinfo");
+    std::string sec_istio_auth_userinfo;
+    // Currently the value in "sec-istio-auth-userinfo" header is a string of
+    // base64url-encoded payload JSON. Decode payload to populate auth
+    // attributes.
+    // TODO: Support extracting auth attributes from all types of contents in
+    // "sec-istio-auth-userinfo" header.
+    if (FindHeaderByName(kSecIstioAuthUserinfo, &sec_istio_auth_userinfo)) {
+      Json::ObjectSharedPtr json_object =
+          Auth::Jwt::LoadFromBase64UrlEncodedPayload(sec_istio_auth_userinfo);
+      if (json_object != nullptr) {
```

Comment:
Use JsonObje::iterate() function to get all map elements,  users can define any map elements as they want.  we have to extract all of them.